packages/hadron-type-checker/src/type-checker.js

```diff
@@ -343,6 +344,9 @@ class TypeChecker {
       if (object._bsontype === OBJECT_ID) {
         return 'ObjectId';
       }
+      if (object._bsontype === SYMBOL) {
+        return 'BSONSymbol';
+      }
```

Comment:
Alternatively we can change type checker map to have Symbol instead of BSONSymbol, but this seems more aligned with how this package works right now